src/libpsl-native/src/getcommonlstat.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! @brief returns the stat of a file
+
+
+#include <errno.h>
+#include <assert.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <pwd.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <stdio.h>
+
+#include "getcommonlstat.h"
+
+// Provide a common structure for the various different stat structures.
+// This should be safe to call on all platforms
+int GetCommonLStat(const char* path, struct CommonStat* commonStat)
```

Comment:
Since in this project, all the APIs have a 1 to 1 mapping with the native APIs, we should have two APIs namely `GetCommonStat` -> `stat` and `GetCommonLStat` -> `lstat`